web/src/copy_and_paste.ts

```diff
@@ -402,6 +402,23 @@ export function paste_handler_converter(paste_html: string): string {
         .parseFromString(paste_html, "text/html")
         .querySelector("body");
     assert(copied_html_fragment !== null);
+
+    const para_elements = copied_html_fragment.querySelectorAll("p");
+
+    for (const element of para_elements) {
+        const children = [...element.childNodes];
+        for (const child of children) {
+            // Check if the current node is a text node
+            if (child.nodeType === Node.TEXT_NODE) {
+                // Create a new span for the text node
```

Comment:
These comments aren't helpful. Instead, do a block comment just before the loop explaining what the algorithm is doing. Something like:

> The below logic for pasting KaTeX cannot handle text nodes directly inside paragraphs, because ... . To work around that limitation, we wrap text nodes inside paragraphs with spans, which avoids that problem because ....